src/components/pages/public/AllMoviesPage.vue

```diff
@@ -1,11 +1,89 @@
 <template>
-  <div>This is all movies page.</div>
+  <div class="all-movies-page">
+    <BreadcrumbNav class="all-movies-page__breadcrumb-nav" />
+    <h2 class="all-movies-page__heading">All the movies</h2>
+    <div class="all-movies-page__wrapper">
+      <div class="all-movies-page__search-wrapper">
+        <BaseInput
+          v-model="query"
+          class="all-movies-page__search-input"
+          type="search"
+          placeholder="What are you looking for?"
+          label="search"
+        />
+      </div>
+      <MovieSelect />
+    </div>
+    <div class="all-movies-page__movies-list">
+      <MovieCard v-for="movie in movies" :movie="movie" :key="movie.id" />
+    </div>
+  </div>
 </template>
 
 <script>
+import BreadcrumbNav from "@/components/BreadcrumbNav.vue";
+import BaseInput from "@/components/base/BaseInput.vue";
+import MovieSelect from "@/components/base/MovieSelect.vue";
+import MovieCard from "@/components/MovieCard.vue";
+
 export default {
   name: "AllMoviesPage",
+  components: { BreadcrumbNav, BaseInput, MovieSelect, MovieCard },
+  data() {
+    return {
+      query: "",
+    };
+  },
+  computed: {
+    movies() {
+      return this.$store.state.movies;
```

Comment:
In real life application searching/filtering is mostly done on API side so you need to pass some additional parameters to the request BUT since there is no pagination it is ok to implement filtering here.

Look you have the `query` in state so you know what the user is searching for, you have the list of movies from API. Can you filter those `movies` according to the `query` with `.filter` method? (select only those movies whose titles contain the search term; remember that search should be case-insensitive)

🤞